src/custom/pages/Claim/Investment/InvestOption.tsx

```diff
@@ -0,0 +1,157 @@
+`
+
+const RangeStep = styled.button`
+  background: none;
+  border: none;
+  font-size: 0.8rem;
+  cursor: pointer;
+  color: blue;
+  padding: 0;
+`
+
+export default function InvestOption({ approveState, approveCallback, updateInvestAmount, claim }: InvestOptionProps) {
+  const { currencyAmount, price, cost: maxCost, investedAmount } = claim
+
+  const { account } = useActiveWeb3React()
+
+  const token = currencyAmount?.currency
+
+  const balance = useCurrencyBalance(account || undefined, token)
+
+  const handlePercentChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    console.log(event.target.value)
+  }
+
+  const handleStepChange = (value: number) => {
+    console.log(value)
+  }
+
+  const onMaxClick = useCallback(() => {
+    if (!maxCost || !balance) {
+      return
+    }
+
+    const amount = maxCost.greaterThan(balance) ? balance : maxCost
+    // store the value as a string to prevent unnecessary re-renders
+    const investAmount = formatUnits(amount.quotient.toString(), balance.currency.decimals)
+
+    updateInvestAmount(claim.index, investAmount)
+  }, [balance, claim.index, maxCost, updateInvestAmount])
+
+  const vCowAmount = useMemo(() => {
+    if (!token || !price) {
+      return
+    }
+
+    const investA = CurrencyAmount.fromRawAmount(token, parseUnits(investedAmount, token.decimals).toString())
+    return investA.multiply(price)
+  }, [investedAmount, price, token])
+
+  return (
+    <InvestTokenGroup>
+      <div>
+        <span>
+          <TokenLogo symbol={currencyAmount?.currency?.symbol || '-'} size={72} />
+          <CowProtocolLogo size={72} />
+        </span>
+        <h3>Buy vCOW with {currencyAmount?.currency?.symbol}</h3>
+      </div>
+
+      <span>
+        <InvestSummary>
+          <span>
+            <b>Price</b>{' '}
+            <i>
+              {formatSmart(price)} vCoW per {currencyAmount?.currency?.symbol}
+            </i>
+          </span>
+          <span>
+            <b>Token approval</b>
+            <i>
+              {approveState === ApprovalState.NOT_APPROVED ? (
+                `${currencyAmount?.currency?.symbol} not approved`
+              ) : (
+                <Row>
+                  {currencyAmount?.currency?.symbol} approved{' '}
+                  <CheckCircle color="lightgreen" style={{ marginLeft: 5 }} />
+                </Row>
+              )}
+            </i>
+            {approveState === ApprovalState.NOT_APPROVED && (
+              <button onClick={approveCallback}>Approve {currencyAmount?.currency?.symbol}</button>
+            )}
+          </span>
+          <span>
+            <b>Max. investment available</b>{' '}
+            <i>
+              {formatSmart(maxCost) || '0'} {currencyAmount?.currency?.symbol}
+            </i>
+          </span>
+          <span>
+            <b>Available investment used</b>
+
+            <div>
+              <RangeSteps>
+                {[0, 25, 50, 75, 100].map((step: number) => (
```

Comment:
this array should probably just be created outside the body

Reply:
You can leave this part out, there's a component (semi-)ready coming from Protofire